### PR TITLE
Initialise UI gems.

### PR DESCRIPTION
**Pull request by lowjoel**
- Use jQuery from a CDN (vs baking it into our application.js)
 - Use Bootstrap + Bootstrap-sass-extras
 - Add themes for instances
 - Allow themes to selectively override other views

### CODE REVIEW

**Comment by lowjoel** [comment]
@jsyeo after this is merged, I think we got to create another issue to get us to replace my local themes_on_rails fork with the upstream one. I can't use the upstream themes_on_rails because of yoolk/themes_on_rails#16

@wangqiang1208 any comments?

File: app/themes/default/views/layouts/default.html.slim [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                = link_to('Logout', destroy_user_session_path, method: :delete)
+            - else
+              li
+                = link_to('Sign Up', new_user_registration_path)
+              li
+                = link_to('Login', new_user_session_path)
+
+    div.container-fluid
+      = flash_messages
+      div class=page_class
```
Comment (jsyeo) [inline app/themes/default/views/layouts/default.html.slim]: Hmm, should this be `div.page_class`?

Comment (lowjoel) [inline app/themes/default/views/layouts/default.html.slim]: this is correct. I want to call the `page_class` helper and set the class of the div to that. equivalent to (in erb):

```erb
<div class="<%= page_class %>">...
```

Comment (jsyeo) [inline app/themes/default/views/layouts/default.html.slim]: :ok_hand:

**Pull request merged**